torch/tensor.py

```diff
@@ -722,4 +722,13 @@ def _update_names(self, names, inplace):
         else:
             return super(Tensor, self).rename(names)
 
+    @property
+    def grad(self):
+        if not self.requires_grad:
+            warnings.warn("Accessing .grad on a Tensor that does not require gradients always returns None.")
+        if not hasattr(self, "retains_grad") and not self.is_leaf:
```

Comment:
wait, do both this and retain_grad above mean that I can't actually turn retain_grad off?

Reply:
Given the implementation above, the hook cannot be removed.
So no we cannot remove the retain_grad. But that's fine as this is only used on temporary Tensors during the forward.